### PR TITLE
Add code projects

### PR DESCRIPTION
**Pull request by laptou**
- Add some information about the code projects I've done to the page.
- Refactor the code section and cooking section.

Screenshot 2020-06-03 at 17 25 34 - Display 1

### CODE REVIEW

File: portfolio/src/main/client/src/components/controls/readmore.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    htmlElement`<a href="javascript: void 0" class="expander">see more</a>`;
+
+  expanderEl.addEventListener('click', () => {
+    if (!contentEl.classList.contains('collapsed')) {
+      contentEl.classList.add('collapsed');
+      contentEl.style.maxHeight = '';
+
+      expanderEl.innerText = 'see more';
+    } else {
+      contentEl.style.maxHeight = `${contentEl.scrollHeight}px`;
```
Comment (jpnickolas) [inline portfolio/src/main/client/src/components/controls/readmore.ts:26]: nit: add a comment about why you need this max height set in javascript. It's unclear at first glance that this is to preserve the css transition.

File: portfolio/src/main/client/src/components/sections/project.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import {htmlElement} from '@src/util/html';
+import {LightboxItem} from '../controls/lightbox';
+import {ResponsiveImageInfo} from '../controls/responsive-image';
+import {ReadMore} from '../controls/readmore';
+
+export interface ProjectInfo {
```
Comment (jpnickolas) [inline portfolio/src/main/client/src/components/sections/project.ts:9]: Any exported item should have a jsdoc since it's part of the project level API.

**Review by jpnickolas — CHANGES_REQUESTED** [review]

File: portfolio/src/main/client/src/components/controls/readmore.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {htmlElement, Renderable} from '@src/util/html';
+import '@res/style/controls/readmore.scss';
+
+export const ReadMore = (content: Renderable): HTMLElement => {
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/components/controls/readmore.ts:12]: nit: add JSDoc comments for all exported items

File: portfolio/src/main/client/src/components/sections/cooking.ts [hunk 4]
```diff
@@ -7,14 +7,34 @@ export interface DishInfo {
...
+// front matter on property `attributes` and the HTML corresponding to the
+// markdown on property `html`.
+
+// type assertion b/c webpack context returns `any` since it does not know
+// the type of the modules
+const dishInfos = dishContext
+  .keys()
+  .map((key) => dishContext(key) as DishInfo);
+
+export const CookingSection = (): HTMLElement =>
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/components/sections/cooking.ts]: nit: add JSDoc comments for all exported items (same with others in this file)

File: portfolio/src/main/client/src/components/sections/project.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+   */
+  html: string;
+}
+
+const projectContext = require.context('@res/text/code', false);
+const projectInfos = projectContext
+  .keys()
+  .map((key) => ({key, info: projectContext(key) as ProjectInfo}));
+
+export const ProjectSection = (): HTMLElement =>
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/components/sections/project.ts:50]: nit: add JSDoc comments for all exported items (same with others in this file)

File: portfolio/src/main/client/src/util/html.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import type {Arrunk} from './types';
 
-type Renderable = Arrunk<string | Node | null | undefined>;
+export type Renderable = Arrunk<string | Node | null | undefined>;
```
Comment (jessicaslaughter) [inline portfolio/src/main/client/src/util/html.ts:7]: nit: add JSDoc comments for all exported items

**Review by jessicaslaughter — CHANGES_REQUESTED** [review]
lgtm, just a few more comments needed for any remaining uncommented exports

File: portfolio/src/main/client/src/components/sections/cooking.ts (hunk 4 above)
Comment (laptou) [reply portfolio/src/main/client/src/components/sections/cooking.ts]: What meaningful documentation can I add to this element? `/** This is the cooking section. */` seems pretty useless, but it doesn't do anything except display the cooking section.

**Review by jessicaslaughter — APPROVED** [review]

**Review by jpnickolas — APPROVED** [review]

**Pull request merged**